qiskit/pulse/builder.py

```diff
@@ -0,0 +1,1986 @@
+            dt = active_backend().configuration().dt
+            accumulated_phase = duration * dt * frequency % (2*np.pi)
+            shift_phase(channel, -accumulated_phase)
+        shift_frequency(channel, -frequency)
+
+
+# Channels
+def drive_channel(qubit: int) -> channels.DriveChannel:
+    """Return ``DriveChannel`` for ``qubit`` on the active builder backend.
+
+    Examples:
+
+    .. jupyter-execute::
+
+        from qiskit import pulse
+        from qiskit.test.mock import FakeOpenPulse2Q
+
+        backend = FakeOpenPulse2Q()
+
+        with pulse.build(backend):
+            assert pulse.drive_channel(0) == pulse.DriveChannel(0)
+
+    .. note:: Requires the active builder context to have a backend set.
+    """
+    return active_backend().configuration().drive(qubit)
+
+
+def measure_channel(qubit: int) -> channels.MeasureChannel:
+    """Return ``MeasureChannel`` for ``qubit`` on the active builder backend.
+
+    Examples:
+
+    .. jupyter-execute::
+
+        from qiskit import pulse
+        from qiskit.test.mock import FakeOpenPulse2Q
+
+        backend = FakeOpenPulse2Q()
+
+        with pulse.build(backend):
+            assert pulse.measure_channel(0) == pulse.MeasureChannel(0)
+
+    .. note:: Requires the active builder context to have a backend set.
+    """
+    return active_backend().configuration().measure(qubit)
+
+
+def acquire_channel(qubit: int) -> channels.AcquireChannel:
+    """Return ``AcquireChannel`` for ``qubit`` on the active builder backend.
+
+    Examples:
+
+    .. jupyter-execute::
+
+        from qiskit import pulse
+        from qiskit.test.mock import FakeOpenPulse2Q
+
+        backend = FakeOpenPulse2Q()
+
+        with pulse.build(backend):
+            assert pulse.acquire_channel(0) == pulse.AcquireChannel(0)
+
+    .. note:: Requires the active builder context to have a backend set.
+    """
+    return active_backend().configuration().acquire(qubit)
+
+
+def control_channels(*qubits: Iterable[int]) -> List[channels.ControlChannel]:
+    """Return ``AcquireChannel`` for ``qubit`` on the active builder backend.
+
+    Return the secondary drive channel for the given qubit -- typically
+    utilized for controlling multi-qubit interactions.
+
+    Examples:
+
+    .. jupyter-execute::
+
+        from qiskit import pulse
+        from qiskit.test.mock import FakeOpenPulse2Q
+
+        backend = FakeOpenPulse2Q()
+        with pulse.build(backend):
+            assert pulse.control_channels(0, 1) == [pulse.ControlChannel(0)]
+
+    .. note:: Requires the active builder context to have a backend set.
+
+    Args:
+      qubits: Tuple or list of ordered qubits of the form
+        `(control_qubit, target_qubit)`.
+
+    Returns:
+        List of control channels associated with the supplied ordered list
+        of qubits.
+    """
+    return active_backend().configuration().control(qubits=qubits)
+
+
+# Base Instructions
+def delay(duration: int,
+          channel: channels.Channel):
```

Comment:
I'm not sure this is that simple. `delay_qubits` is special because qubits might share a channel and therefore to properly delay across multiple qubits without inserting duplicate delays we want to delay only on the union of the qubits. Changing delay to accept qubits would mean accepting multiple channels for a builder instruction that otherwise maps to an atomic instruction (on a channel).